app/src/main/java/ch/sdp/vibester/activity/DeleteSongsActivity.kt

```diff
@@ -0,0 +1,145 @@
+        deleteButton.id = iterator //resources.getIdentifier("id_$trimmed", "id", packageName)
+        deleteButton.text = trimmed
+        deleteButton.gravity = Gravity.CENTER
+        deleteButton.layoutParams = LayoutParams(MATCH_PARENT, WRAP_CONTENT)
+        deleteButton.setOnClickListener {
+            deleteDownloadedSong(it, layout)
+        }
+        return deleteButton
+    }
+
+    /*
+     * Handles the deletion of a downloaded extract. Also handles the removal of the said extract
+     * from the text file which keeps track of all downloads.
+     */
+    private fun deleteDownloadedSong(btn: View, layout: LinearLayout): Boolean {
+        var records = File(applicationContext.getExternalFilesDir(Environment.DIRECTORY_DOWNLOADS), "records.txt")
+        var tempRecords = File(applicationContext.getExternalFilesDir(Environment.DIRECTORY_DOWNLOADS), "temp.txt")
+
+        var recordReader = BufferedReader(FileReader(records))
+        var recordWriter = BufferedWriter(FileWriter(tempRecords, true))
+
+        var currentLine = recordReader.readLine()
+        var button: Button = btn as Button
+
+        while(currentLine != null) {
+            var trimmed = currentLine.trim()
+
+            if(trimmed == button.text.toString()) {
+                currentLine = recordReader.readLine()
+                continue
+            }
+
+            recordWriter.write(currentLine)
+            recordWriter.newLine()
+            currentLine = recordReader.readLine()
+        }
+
+        recordWriter.close()
+        recordReader.close()
+
+        if(tempRecords.length() == 0L) {
+            createNoSongsView(layout)
+        }
+
+        records.delete()
+        if(tempRecords.renameTo(records)) {
+            layout.removeView(btn)
+            var songToDelete = File(applicationContext.getExternalFilesDir(Environment.DIRECTORY_DOWNLOADS), "extract_of_${button.text}")
+            if(songToDelete.delete()) {
+                Toast.makeText(applicationContext, "Song successfully removed!", Toast.LENGTH_LONG).show()
+                return true
+            }
+        }
+        Toast.makeText(applicationContext, "Song was unable to be removed!", Toast.LENGTH_LONG).show()
+        return false
+    }
+
+    /*
+     * Handles the creation and initialization of a TextView which is then added to the layout.
+     */
+    private fun createNoSongsView(layout: LinearLayout) {
+        val noDownloadsText = TextView(this)
+        noDownloadsText.id = resources.getIdentifier("noDownloadsView", "id", packageName)
+        noDownloadsText.text = resources.getString(R.string.delete_nothing_to_delete)
+        noDownloadsText.gravity = Gravity.CENTER
+        noDownloadsText.layoutParams = LinearLayout.LayoutParams(MATCH_PARENT, MATCH_PARENT)
+        layout.addView(noDownloadsText)
+    }
+
+    fun switchToWelcome(view: View) {
+        sendDirectIntent(WelcomeActivity::class.java)
+    }
+
+    private fun sendDirectIntent(arg: Class<*>?) {
```

Comment:
Don't know if you want to do it now, but we have an IntentSwitcher file that is here to store helper of that type. According to codeclimate there are 4 duplication in whole project. We may consider refactoring that part.

Reply:
Done, using IntentSwitcher on all three of the welcome page ones. Reverted the other one back to what it was normally, let's hope cc doesn't complain anymore lol